src/Fusonic/Linq/Iterator/ChunkIterator.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of Fusonic-linq.
+ * https://github.com/fusonic/fusonic-linq
+ *
+ * (c) Fusonic GmbH
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Fusonic\Linq\Iterator;
+
+use Countable;
+use Fusonic\Linq\Linq;
+use Iterator;
+
+/**
+ * Iterates over an iterator, returning Linq objects of the given chunk size.
+ */
+class ChunkIterator implements Iterator, Countable
+{
+    /**
+     * @var Iterator
+     */
+    private $iterator;
+
+    /**
+     * @var int
+     */
+    private $i = 0;
+
+    /**
+     * @var int
+     */
+    private $chunkSize;
+
+    public function __construct(Iterator $iterator, $chunkSize)
+    {
+        $this->iterator = $iterator;
+        $this->chunkSize = $chunkSize;
+    }
+
+    /**
+     * @return Linq
+     * @todo Should this chunking logic stay here? This might be better in `next()` but then `current()` should return
```

Comment:
Yes, this code belongs into the "next" method.
You then need a private $chunk member, which holds the current chunk. Subsequent calls to "current()" must always return the same value and must not mutate the state.